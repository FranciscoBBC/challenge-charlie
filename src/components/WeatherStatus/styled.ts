import styled, { css } from 'styled-components';

export const WeatherHolder = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    flex: 8;
    width: 100%;
    height: auto;
`;

export const Grid = styled.div`
    display: grid;
    grid-template-columns: 4fr 3fr;
    grid-template-rows: 3fr 1fr 1fr;
    width: 100%;
    height: 100%;
`;

// today weather
export const statusWrapper = styled.div`
    display: flex;
    flex-direction: column;
    opacity: 85%;
    height: 100%;
    background: #edb915;
    padding: 0 5px 0 5px;
    justify-content: center;
    @media screen and (max-width: 768px) {
        p:first-child {
            padding: 0;
        }
    }
`;

export const TomorrowWrapper = styled.div`
    display: flex;
    flex-direction: column;
    width: 100%;
    background: #facc05;
    opacity: 93%;
    padding: 0 5px 0 5px;
    @media screen and (max-width: 425px) {
        p {
            font-size: 18px;
        }
    }
`;

export const AfterTomorrowWrapper = styled.div`
    display: flex;
    flex-direction: column;
    width: 100%;
    background: #b89503;
    padding: 0 5px 0 5px;
    @media screen and (max-width: 425px) {
        p {
            font-size: 18px;
        }
    }
`;

type IconWrapperProps = {
    bgColor: string;
    opacity?: string;
};

export const IconWrapper = styled.div<IconWrapperProps>`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100%;
    background: ${({ bgColor }) => bgColor};
    opacity: ${({ opacity }) => opacity || '1'};
`;

type textProps = {
    fontSize?: string;
    fontWeight?: string;
    fontFamily?: string;
    padding?: string;
    breakpoints?: {
        mobile?: {
            fontSize?: string;
            fontWeight?: string;
        };
    };
};

export const Text = styled.p<textProps>`
    color: #fff;
    font-size: ${({ fontSize }) => fontSize || '14px'};
    font-weight: ${({ fontWeight }) => fontWeight || '400'};
    font-family: ${({ fontFamily }) =>
        fontFamily ||
        "-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif"};
    padding: ${({ padding }) => padding || 0};
    ${({ breakpoints }) =>
        breakpoints?.mobile &&
        css`
            @media screen and (max-width: 768px) {
                font-size: ${breakpoints?.mobile?.fontSize || '14px'};
                font-weight: ${breakpoints?.mobile?.fontWeight || '400'};
            }
        `};
`;
