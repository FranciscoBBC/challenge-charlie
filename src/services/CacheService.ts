class CacheService {
    private static minutsToTTL = (ttl: number) => {
        const newTTL = Date.now() + ttl * 60000;
        return new Date(newTTL).toUTCString();
    };

    public static setCookie = (key: string, data: string, ttl?: number) => {
        let timeToLive = ttl;
        if (!timeToLive) timeToLive = 30;
        document.cookie = `${key}=${data}; expires=${this.minutsToTTL(
            timeToLive
        )};`;
    };

    public static getCookie = (key: string): string => {
        const name = `${key}=`;
        const decodedCookie = decodeURIComponent(document.cookie);
        const cookieArray = decodedCookie.split(';');

        for (let i = 0; i < cookieArray.length; i += 1) {
            let currentCookie = cookieArray[i];
            while (currentCookie.charAt(0) === ' ') {
                currentCookie = currentCookie.substring(1);
            }
            if (currentCookie.indexOf(name) === 0) {
                return currentCookie.substring(
                    name.length,
                    currentCookie.length
                );
            }
        }
        return '';
    };
}

export default CacheService;
