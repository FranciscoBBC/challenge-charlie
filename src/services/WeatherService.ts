/* eslint-disable camelcase */
import { apiInstance } from './ConfigService';

class WeatherService {
    private static openWeatherBaseRoute =
        'https://api.openweathermap.org/data/2.5';

    private static openWeatherApiKey = '7ba73e0eb8efe773ed08bfd0627f07b8';

    public static getWeatherByCityName = async (
        cityName: string
    ): Promise<getWeatherByCityNameReturn | null> => {
        try {
            const response = (await apiInstance.get(
                `${this.openWeatherBaseRoute}/weather`,
                {
                    params: {
                        q: cityName,
                        appid: this.openWeatherApiKey,
                        units: 'metric',
                        lang: 'pt_br'
                    }
                }
            )) as any;
            if (response.data?.cod) return null;

            const { weather, main, wind } = await response.data;
            return { weather, main, wind } as getWeatherByCityNameReturn;
        } catch (err) {
            throw new Error(
                'ocorreu um erro ao buscar as informações de clima atual'
            );
        }
    };

    public static getWeatherNextDays = async (
        lat: number,
        lon: number
    ): Promise<returnDailyWeather | null> => {
        try {
            const response = (await apiInstance.get(
                `${this.openWeatherBaseRoute}/onecall`,
                {
                    params: {
                        lat,
                        lon,
                        exclude: 'minutely,hourly,alerts',
                        appid: this.openWeatherApiKey,
                        units: 'metric',
                        lang: 'pt_br'
                    }
                }
            )) as any;
            if (response.data?.cod) return null;
            const { daily, current } = response.data as getWeatherForNextDays;
            return {
                today: current,
                tomorrow: daily[1],
                afterTomorrow: daily[2]
            };
        } catch (err) {
            throw new Error(
                'ocorreu um erro ao buscar as informações de clima'
            );
        }
    };
}

type todayWeather = {
    temp: number;
    pressure: number;
    humidity: number;
    wind_speed: number;
    wind_deg: number;
    weather: {
        main:
            | 'Atmosphere'
            | 'Snow'
            | 'Rain'
            | 'Thunderstorm'
            | 'Drizzle'
            | 'Clouds'
            | 'Clear'
            | 'Fog';
        description: string;
    }[];
};

type dailyWeather = {
    temp: {
        day: number;
    };
    pressure: number;
    humidity: number;
    wind_speed: number;
    wind_deg: number;
    weather: {
        main:
            | 'Atmosphere'
            | 'Snow'
            | 'Rain'
            | 'Thunderstorm'
            | 'Drizzle'
            | 'Clouds'
            | 'Clear';
        description: string;
    }[];
};

type getWeatherForNextDays = {
    current: todayWeather;
    daily: dailyWeather[];
};

export type returnDailyWeather = {
    today: todayWeather;
    tomorrow: dailyWeather;
    afterTomorrow: dailyWeather;
};

type getWeatherByCityNameReturn = {
    weather: {
        id: number;
        main: string;
        description: string;
        icon: string;
    }[];
    main: {
        temp: number;
        feels_like: number;
        temp_min: number;
        temp_max: number;
        pressure: number;
        humidity: number;
    };
    wind: {
        speed: number;
        deg: number;
    };
};

export default WeatherService;
