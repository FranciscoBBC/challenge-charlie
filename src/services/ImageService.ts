import { apiInstance } from 'services/ConfigService';

class ImageService {
    public static getImageFromBing = async () => {
        try {
            const response = (await apiInstance.get(
                'https://cors-anywhere.herokuapp.com/https://www.bing.com/HPImageArchive.aspx?format=js&idx=0&n=1&mkt=pt-BR'
            )) as any;

            const { images } = response.data;
            if (images[0]?.url) return `https://www.bing.com${images[0].url}`;
            return undefined;
        } catch (exception) {
            console.error(exception);
            return undefined;
        }
    };
}

export default ImageService;
