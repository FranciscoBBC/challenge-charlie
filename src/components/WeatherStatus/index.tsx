import React, { useMemo, useState, useCallback } from 'react';
import { FiSun } from 'react-icons/fi';
import {
    BsCloudSnow,
    BsCloudRain,
    BsCloudLightningRain,
    BsCloudDrizzle,
    BsClouds,
    BsCloudFog2
} from 'react-icons/bs';
import { WiCloudyWindy } from 'react-icons/wi';
import { returnDailyWeather } from 'services/WeatherService';
import * as S from './styled';

type WeatherStatusProps = {
    dailyWeather: returnDailyWeather | null;
};

type TemperatureTypes = 'celsius' | 'fahrenheit';

const WeatherStatus = ({ dailyWeather }: WeatherStatusProps) => {
    const [temperatureMeasurement, setTemperatureMeasurement] =
        useState<TemperatureTypes>('celsius');

    // retorna o status do clima atual (nublado, chovendo...)
    const description = useMemo(() => {
        const weatherDescription = dailyWeather?.today.weather[0].description;
        if (!weatherDescription) return null;
        return (
            weatherDescription.charAt(0).toUpperCase() +
            weatherDescription.slice(1)
        );
    }, [dailyWeather?.today.weather]);

    // retorna velocidade do vento
    const wind = useMemo(() => {
        const directions = ['N', 'NE', 'L', 'SE', 'S', 'SO', 'O', 'NO'];
        let deg = dailyWeather?.today.wind_deg;
        if (deg) {
            deg += 22.5;

            if (deg < 0) deg = 360 - (Math.abs(deg) % 360);
            else deg %= 360;
            const position = Math.floor(deg / 45);

            const windPosition = directions[position];
            return `${windPosition} ${dailyWeather?.today.wind_speed}Km/h`;
        }
        return '';
    }, [dailyWeather?.today.wind_deg, dailyWeather?.today.wind_speed]);

    // retorna temperatura no formato correto
    const celsiusToFahrenheit = useCallback(
        temperature => {
            if (!temperature) return '0ºC';
            if (temperatureMeasurement === 'celsius')
                return `${Math.floor(temperature)}ºC`;

            return `${Math.floor((temperature * 9) / 5 + 32)}ºF`;
        },
        [temperatureMeasurement]
    );

    // retorna o icone referente ao clima atual
    const weatherIcon = useMemo(() => {
        const icon = dailyWeather?.today.weather[0].main;
        const iconList = {
            Atmosphere: <WiCloudyWindy color="#fff" size="80%" />,
            Snow: <BsCloudSnow color="#fff" size="80%" />,
            Rain: <BsCloudRain color="#fff" size="80%" />,
            Thunderstorm: <BsCloudLightningRain color="#fff" size="80%" />,
            Drizzle: <BsCloudDrizzle color="#fff" size="80%" />,
            Clouds: <BsClouds color="#fff" size="80%" />,
            Clear: <FiSun color="#fff" size="80%" />,
            Fog: <BsCloudFog2 color="#fff" size="80%" />,
            Mist: <BsCloudFog2 color="#fff" size="80%" />
        };
        if (icon) return iconList[icon];
        return null;
    }, [dailyWeather?.today.weather]);

    // define a cor de fundo dependendo da temperatura atual
    const backgroundColor = useMemo(() => {
        if (!dailyWeather?.today.temp)
            return {
                one: '#606060',
                two: '#3D3D3D',
                three: '#262626'
            };

        if (dailyWeather?.today.temp < 15)
            return {
                one: '#6497b1',
                two: '#005b96',
                three: '#011f4b'
            };
        if (dailyWeather?.today.temp > 35)
            return {
                one: '#ff5252',
                two: '#ff0000',
                three: '#a70000'
            };

        return {
            one: '#EDB915',
            two: '#facc05',
            three: '#B79404'
        };
    }, [dailyWeather?.today.temp]);

    return (
        <S.WeatherHolder>
            <S.Grid>
                <S.IconWrapper
                    data-testid="weather-icon"
                    bgColor={backgroundColor.one}
                    opacity="85%"
                >
                    {weatherIcon}
                </S.IconWrapper>
                <S.statusWrapper
                    data-testid="today-weather"
                    bgColor={backgroundColor.one}
                >
                    <S.Text
                        fontSize="1.8rem"
                        padding="8px 0 3px 0"
                        breakpoints={{ mobile: { fontSize: '1.5rem' } }}
                    >
                        HOJE
                    </S.Text>
                    <S.Text
                        data-testid="temperature"
                        cursor="pointer"
                        fontSize="1.8rem"
                        fontFamily="'Open Sans'"
                        breakpoints={{ mobile: { fontSize: '1.5rem' } }}
                        onClick={() =>
                            setTemperatureMeasurement(temperatureType =>
                                temperatureType === 'celsius'
                                    ? 'fahrenheit'
                                    : 'celsius'
                            )
                        }
                    >
                        {celsiusToFahrenheit(dailyWeather?.today.temp)}
                    </S.Text>
                    <S.Text
                        fontSize="2rem"
                        padding="16px 0 10px 0"
                        breakpoints={{ mobile: { fontSize: '1.8rem' } }}
                    >
                        {description}
                    </S.Text>
                    <S.Text
                        data-testid="wind"
                        fontSize="1.4rem"
                        breakpoints={{ mobile: { fontSize: '1.2rem' } }}
                    >
                        Vento: {wind}
                    </S.Text>
                    <S.Text
                        data-testid="humidity"
                        fontSize="1.4rem"
                        breakpoints={{ mobile: { fontSize: '1.2rem' } }}
                    >
                        Humidade: {dailyWeather?.today.humidity}%
                    </S.Text>
                    <S.Text
                        data-testid="pressure"
                        fontSize="1.4rem"
                        breakpoints={{ mobile: { fontSize: '1.2rem' } }}
                    >
                        Pressão: {dailyWeather?.today.pressure}hPA
                    </S.Text>
                </S.statusWrapper>
                <S.IconWrapper bgColor={backgroundColor.two} opacity="93%" />
                <S.TomorrowWrapper
                    data-testid="tomorrow-weather"
                    bgColor={backgroundColor.two}
                >
                    <S.Text fontSize="1.5rem" padding="8px 0 3px 0">
                        AMANHÃ
                    </S.Text>
                    <S.Text fontSize="1.5rem" fontFamily="'Open Sans'">
                        {celsiusToFahrenheit(dailyWeather?.tomorrow.temp.day)}
                    </S.Text>
                </S.TomorrowWrapper>
                <S.IconWrapper bgColor={backgroundColor.three} />
                <S.AfterTomorrowWrapper
                    data-testid="aftertomorrow-weather"
                    bgColor={backgroundColor.three}
                >
                    <S.Text fontSize="1.5rem" padding="8px 0 3px 0">
                        DEPOIS DE AMANHÃ
                    </S.Text>
                    <S.Text fontSize="1.5rem" fontFamily="'Open Sans'">
                        {celsiusToFahrenheit(
                            dailyWeather?.afterTomorrow.temp.day
                        )}
                    </S.Text>
                </S.AfterTomorrowWrapper>
            </S.Grid>
        </S.WeatherHolder>
    );
};

export default WeatherStatus;
