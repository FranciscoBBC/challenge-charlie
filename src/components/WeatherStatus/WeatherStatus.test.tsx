import React from 'react';
import { render, screen } from '@testing-library/react';
import { returnDailyWeather } from 'services/WeatherService';
import userEvent from '@testing-library/user-event';
import WeatherStatus from '.';

describe('<WeatherStatus />', () => {
    const mockData = {
        today: {
            temp: 12,
            pressure: 1019,
            humidity: 77,
            wind_speed: 3.13,
            wind_deg: 200,
            weather: [
                {
                    main: 'Rain',
                    description: 'chuva forte'
                }
            ]
        },
        tomorrow: {
            temp: {
                day: 30
            },
            pressure: 123,
            humidity: 456,
            wind_speed: 2,
            wind_deg: 4.54,
            weather: [
                {
                    main: 'Clear',
                    description: 'céu limpo'
                }
            ]
        },
        afterTomorrow: {
            temp: {
                day: 35
            },
            pressure: 123,
            humidity: 456,
            wind_speed: 2,
            wind_deg: 4.54,
            weather: [
                {
                    main: 'Clouds',
                    description: 'nublado'
                }
            ]
        }
    } as returnDailyWeather;

    it('should be rendered', () => {
        const { container } = render(<WeatherStatus dailyWeather={mockData} />);
        expect(container.firstChild).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should be rendered without props', () => {
        const { container } = render(<WeatherStatus dailyWeather={null} />);
        expect(container.firstChild).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should have background color', () => {
        render(<WeatherStatus dailyWeather={mockData} />);
        expect(screen.getByTestId('today-weather')).toHaveStyle({
            background: '#6497b1'
        });
        expect(screen.getByTestId('tomorrow-weather')).toHaveStyle({
            background: '#005b96'
        });
        expect(screen.getByTestId('aftertomorrow-weather')).toHaveStyle({
            background: '#011f4b'
        });
    });

    it('should render icon', () => {
        render(<WeatherStatus dailyWeather={mockData} />);
        expect(screen.getByTestId('weather-icon').firstChild).toBeTruthy();
    });

    it('should render today temperature', () => {
        render(<WeatherStatus dailyWeather={mockData} />);
        expect(screen.getByTestId('temperature').firstChild?.nodeValue).toBe(
            '12ºC'
        );
    });

    it('should render tomorow temperature', () => {
        render(<WeatherStatus dailyWeather={mockData} />);
        expect(
            screen.getByTestId('tomorrow-weather').childNodes[1].firstChild
                ?.nodeValue
        ).toBe('30ºC');
    });

    it('should render after tomorow temperature', () => {
        render(<WeatherStatus dailyWeather={mockData} />);
        expect(
            screen.getByTestId('aftertomorrow-weather').childNodes[1].firstChild
                ?.nodeValue
        ).toBe('35ºC');
    });

    it('should render temperature in fahrenheit', () => {
        render(<WeatherStatus dailyWeather={mockData} />);
        const temperatureWrapper = screen.getByTestId('temperature');
        userEvent.click(temperatureWrapper);
        expect(temperatureWrapper.firstChild?.nodeValue).toBe('53ºF');
    });

    it('should render wind', () => {
        render(<WeatherStatus dailyWeather={mockData} />);
        setTimeout(() => {
            expect(screen.getByTestId('wind').firstChild?.nodeValue).toBe(
                'Vento: S 3.13Km/h'
            );
        }, 1000);
    });

    it('should render humidity', () => {
        render(<WeatherStatus dailyWeather={mockData} />);
        setTimeout(() => {
            expect(screen.getByTestId('humidity').firstChild?.nodeValue).toBe(
                'Humidade: 77%'
            );
        }, 1000);
    });

    it('should render pressure', () => {
        render(<WeatherStatus dailyWeather={mockData} />);
        setTimeout(() => {
            expect(screen.getByTestId('pressure').firstChild?.nodeValue).toBe(
                'Pressão: 1019hPA'
            );
        }, 1000);
    });
});
