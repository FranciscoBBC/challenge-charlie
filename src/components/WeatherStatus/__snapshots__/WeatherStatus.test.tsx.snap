// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherStatus /> should be rendered 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 8;
  -ms-flex: 8;
  flex: 8;
  width: 100%;
  height: auto;
}

.c1 {
  display: grid;
  grid-template-columns: 4fr 3fr;
  grid-template-rows: 3fr 1fr 1fr;
  width: 100%;
  height: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  opacity: 85%;
  height: 100%;
  background: #6497b1;
  padding: 0 5px 0 5px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background: #005b96;
  opacity: 93%;
  padding: 0 5px 0 5px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background: #011f4b;
  padding: 0 5px 0 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  background: #6497b1;
  opacity: 85%;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  background: #005b96;
  opacity: 93%;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  background: #011f4b;
  opacity: 1;
}

.c4 {
  cursor: auto;
  color: #fff;
  font-size: 1.8rem;
  font-weight: 400;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 8px 0 3px 0;
}

.c5 {
  cursor: pointer;
  color: #fff;
  font-size: 1.8rem;
  font-weight: 400;
  font-family: 'Open Sans';
  padding: 0;
}

.c6 {
  cursor: auto;
  color: #fff;
  font-size: 2rem;
  font-weight: 400;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 16px 0 10px 0;
}

.c7 {
  cursor: auto;
  color: #fff;
  font-size: 1.4rem;
  font-weight: 400;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 0;
}

.c10 {
  cursor: auto;
  color: #fff;
  font-size: 1.5rem;
  font-weight: 400;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 8px 0 3px 0;
}

.c11 {
  cursor: auto;
  color: #fff;
  font-size: 1.5rem;
  font-weight: 400;
  font-family: 'Open Sans';
  padding: 0;
}

@media screen and (max-width:768px) {
  .c3 p:first-child {
    padding: 0;
  }
}

@media screen and (max-width:425px) {
  .c9 p {
    font-size: 18px;
  }
}

@media screen and (max-width:425px) {
  .c13 p {
    font-size: 18px;
  }
}

@media screen and (max-width:768px) {
  .c4 {
    font-size: 1.5rem;
    font-weight: 400;
  }
}

@media screen and (max-width:768px) {
  .c5 {
    font-size: 1.5rem;
    font-weight: 400;
  }
}

@media screen and (max-width:768px) {
  .c6 {
    font-size: 1.8rem;
    font-weight: 400;
  }
}

@media screen and (max-width:768px) {
  .c7 {
    font-size: 1.2rem;
    font-weight: 400;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      data-testid="weather-icon"
      opacity="85%"
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="80%"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 16 16"
        width="80%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.158 12.025a.5.5 0 0 1 .316.633l-.5 1.5a.5.5 0 0 1-.948-.316l.5-1.5a.5.5 0 0 1 .632-.317zm3 0a.5.5 0 0 1 .316.633l-1 3a.5.5 0 0 1-.948-.316l1-3a.5.5 0 0 1 .632-.317zm3 0a.5.5 0 0 1 .316.633l-.5 1.5a.5.5 0 0 1-.948-.316l.5-1.5a.5.5 0 0 1 .632-.317zm3 0a.5.5 0 0 1 .316.633l-1 3a.5.5 0 1 1-.948-.316l1-3a.5.5 0 0 1 .632-.317zm.247-6.998a5.001 5.001 0 0 0-9.499-1.004A3.5 3.5 0 1 0 3.5 11H13a3 3 0 0 0 .405-5.973zM8.5 2a4 4 0 0 1 3.976 3.555.5.5 0 0 0 .5.445H13a2 2 0 0 1 0 4H3.5a2.5 2.5 0 1 1 .605-4.926.5.5 0 0 0 .596-.329A4.002 4.002 0 0 1 8.5 2z"
        />
      </svg>
    </div>
    <div
      class="c3"
      data-testid="today-weather"
    >
      <p
        class="c4"
        font-size="1.8rem"
      >
        HOJE
      </p>
      <p
        class="c5"
        cursor="pointer"
        data-testid="temperature"
        font-family="'Open Sans'"
        font-size="1.8rem"
      >
        12ºC
      </p>
      <p
        class="c6"
        font-size="2rem"
      >
        Chuva forte
      </p>
      <p
        class="c7"
        data-testid="wind"
        font-size="1.4rem"
      >
        Vento: 
        S 3.13Km/h
      </p>
      <p
        class="c7"
        data-testid="humidity"
        font-size="1.4rem"
      >
        Humidade: 
        77
        %
      </p>
      <p
        class="c7"
        data-testid="pressure"
        font-size="1.4rem"
      >
        Pressão: 
        1019
        hPA
      </p>
    </div>
    <div
      class="c8"
      opacity="93%"
    />
    <div
      class="c9"
      data-testid="tomorrow-weather"
    >
      <p
        class="c10"
        font-size="1.5rem"
      >
        AMANHÃ
      </p>
      <p
        class="c11"
        font-family="'Open Sans'"
        font-size="1.5rem"
      >
        30ºC
      </p>
    </div>
    <div
      class="c12"
    />
    <div
      class="c13"
      data-testid="aftertomorrow-weather"
    >
      <p
        class="c10"
        font-size="1.5rem"
      >
        DEPOIS DE AMANHÃ
      </p>
      <p
        class="c11"
        font-family="'Open Sans'"
        font-size="1.5rem"
      >
        35ºC
      </p>
    </div>
  </div>
</div>
`;

exports[`<WeatherStatus /> should be rendered without props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 8;
  -ms-flex: 8;
  flex: 8;
  width: 100%;
  height: auto;
}

.c1 {
  display: grid;
  grid-template-columns: 4fr 3fr;
  grid-template-rows: 3fr 1fr 1fr;
  width: 100%;
  height: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  opacity: 85%;
  height: 100%;
  background: #606060;
  padding: 0 5px 0 5px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background: #3D3D3D;
  opacity: 93%;
  padding: 0 5px 0 5px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background: #262626;
  padding: 0 5px 0 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  background: #606060;
  opacity: 85%;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  background: #3D3D3D;
  opacity: 93%;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  background: #262626;
  opacity: 1;
}

.c4 {
  cursor: auto;
  color: #fff;
  font-size: 1.8rem;
  font-weight: 400;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 8px 0 3px 0;
}

.c5 {
  cursor: pointer;
  color: #fff;
  font-size: 1.8rem;
  font-weight: 400;
  font-family: 'Open Sans';
  padding: 0;
}

.c6 {
  cursor: auto;
  color: #fff;
  font-size: 2rem;
  font-weight: 400;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 16px 0 10px 0;
}

.c7 {
  cursor: auto;
  color: #fff;
  font-size: 1.4rem;
  font-weight: 400;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 0;
}

.c10 {
  cursor: auto;
  color: #fff;
  font-size: 1.5rem;
  font-weight: 400;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 8px 0 3px 0;
}

.c11 {
  cursor: auto;
  color: #fff;
  font-size: 1.5rem;
  font-weight: 400;
  font-family: 'Open Sans';
  padding: 0;
}

@media screen and (max-width:768px) {
  .c3 p:first-child {
    padding: 0;
  }
}

@media screen and (max-width:425px) {
  .c9 p {
    font-size: 18px;
  }
}

@media screen and (max-width:425px) {
  .c13 p {
    font-size: 18px;
  }
}

@media screen and (max-width:768px) {
  .c4 {
    font-size: 1.5rem;
    font-weight: 400;
  }
}

@media screen and (max-width:768px) {
  .c5 {
    font-size: 1.5rem;
    font-weight: 400;
  }
}

@media screen and (max-width:768px) {
  .c6 {
    font-size: 1.8rem;
    font-weight: 400;
  }
}

@media screen and (max-width:768px) {
  .c7 {
    font-size: 1.2rem;
    font-weight: 400;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      data-testid="weather-icon"
      opacity="85%"
    />
    <div
      class="c3"
      data-testid="today-weather"
    >
      <p
        class="c4"
        font-size="1.8rem"
      >
        HOJE
      </p>
      <p
        class="c5"
        cursor="pointer"
        data-testid="temperature"
        font-family="'Open Sans'"
        font-size="1.8rem"
      >
        0ºC
      </p>
      <p
        class="c6"
        font-size="2rem"
      />
      <p
        class="c7"
        data-testid="wind"
        font-size="1.4rem"
      >
        Vento: 
        
      </p>
      <p
        class="c7"
        data-testid="humidity"
        font-size="1.4rem"
      >
        Humidade: 
        %
      </p>
      <p
        class="c7"
        data-testid="pressure"
        font-size="1.4rem"
      >
        Pressão: 
        hPA
      </p>
    </div>
    <div
      class="c8"
      opacity="93%"
    />
    <div
      class="c9"
      data-testid="tomorrow-weather"
    >
      <p
        class="c10"
        font-size="1.5rem"
      >
        AMANHÃ
      </p>
      <p
        class="c11"
        font-family="'Open Sans'"
        font-size="1.5rem"
      >
        0ºC
      </p>
    </div>
    <div
      class="c12"
    />
    <div
      class="c13"
      data-testid="aftertomorrow-weather"
    >
      <p
        class="c10"
        font-size="1.5rem"
      >
        DEPOIS DE AMANHÃ
      </p>
      <p
        class="c11"
        font-family="'Open Sans'"
        font-size="1.5rem"
      >
        0ºC
      </p>
    </div>
  </div>
</div>
`;
